lib/ui/holiday_list/holiday_list_screen.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_redux/flutter_redux.dart';
+import 'package:holidays/model/fetchable.dart';
+import 'package:holidays/model/holiday_summary.dart';
+import 'package:holidays/redux/app/app_state.dart';
+import 'package:holidays/redux/holiday_list/holiday_list_actions.dart';
+import 'package:holidays/redux/holiday_list/holiday_list_state.dart';
+import 'package:meta/meta.dart';
+import 'package:redux/redux.dart';
+
+class HolidayListScreen extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) => StoreConnector<AppState, _ViewModel>(
+        converter: (Store<AppState> store) => _ViewModel.create(store.state.holidaySummariesState),
+        builder: (BuildContext context, _ViewModel viewModel) => Scaffold(
+              appBar: AppBar(
+                title: Text(viewModel.pageTitle),
+              ),
+              body: _createRefreshWidget(context, viewModel),
+            ),
+      );
+
+  // returns a pull-to-refresh widget that wraps a list of children
+  Widget _createRefreshWidget(BuildContext context, _ViewModel viewModel) {
+    final refreshFunction = () {
+      final refreshAction = FetchHolidaySummariesAction();
+      final store = StoreProvider.of<AppState>(context);
+      store.dispatch(refreshAction);
+      return refreshAction.completer.future;
+    };
+    return RefreshIndicator(child: ListView(children: _createBodyWidgets(context, viewModel)), onRefresh: refreshFunction);
+  }
+
+  // returns a collection of widgets to be displayed in a list view. for the loading/error
+  // states, it will be a list with only one widget in it.
+  List<Widget> _createBodyWidgets(BuildContext context, _ViewModel viewModel) {
```

Comment:
This is super simplistic, but the general gist is that we want to display a list of widgets. In the case of the fetching state, the list contains just a single element. Same for the error state. However, in the success state, we return a list of `Row` objects containing some details about each holiday. Very simple UI elements, but enough to give the idea.